### PR TITLE
Application and Renderer resize improvements

### LINKED ISSUE #6081: Feature request: resizeTo redraw
### Expected Behavior
If a PIXI `Application` has a `resizeTo` element, it would be nice if the element positions inside the application would also update when resizing. Plus some throttling on the resize event would help avoid performance problems.
If this functionality is too complicated to implement (or already present), a best practice example of a responsive application would be very helpful.

### Current Behavior
Only the canvas is resized.

### Environment

- **`pixi.js` version**: 5.1.3

### PR DESCRIPTION
**Pull request by bigtimebuddy**
I've been meaning to tackle this one for awhile, since no one seemed keen on adding the PR for it.

Fixes #6081

### Added

* Introduces `resize` event to the Renderer. Called immediate after view is resized can be used for responsive layouts, or other resizing events.
* Adds throttle resizing on the Application when using `resizeTo`, this will prevent unnecessary extra event calls triggered by the window 'resize'.

### Example

https://jsfiddle.net/bigtimebuddy/u05c1s28/

### CODE REVIEW

**Comment by themoonrat** [comment]
Had a play with a slight modified version of your fiddle: https://jsfiddle.net/w63ahoqd/

Can confirm that no matter how crazy I went on the window resizing, it occurs just ones per render

image

File: packages/app/src/ResizePlugin.ts [hunk 1]
```diff
@@ -9,8 +9,12 @@ import { IApplicationOptions } from './Application';
  */
 export class ResizePlugin
 {
+    public static _resizeId: number;
```
Comment (Zyie) [inline packages/app/src/ResizePlugin.ts]: should `_resizeId`, `_resizeTo`, `queueResize`, and `cancelResize` be private?

`_resizeId` has `_` so it seems private
`_resizeTo` has a get/set
`queueResize`, and `cancelResize` have the `@private` tag in the docs

Comment (bigtimebuddy) [reply packages/app/src/ResizePlugin.ts]: Good question. I was following the lead on the access modifiers, but I guess these can be private. I think the typings for the app plugins need a little more thought.

**Comment by bigtimebuddy** [comment]
Added a feature based on feedback from @GoodBoyDigital to set the amount of time to throttle instead of just using RAF. I agree this is more useful for producing more stable results, especially if you're doing something expensive on resize.

**Comment by bigtimebuddy** [comment]
Uncertain if we should set a default for `resizeThrottle`, since `100` seems more reasonable than `0`, but wonder what other folks think?

**Comment by ivanpopelyshev** [comment]
Usually, people use `setInterval`. Of course your solution is better, but it involves many lines of code.

**Comment by themoonrat** [comment]
The issue I have with this latest change is that if the screen keeps resizing, then nothing is ever updated.
https://jsfiddle.net/opxhkytg/1/

This has a 1 second resizeThrottle setting. Keep slowly resizing the window, and you'll notice that the resize event _never_ kicks in, since a new window resize keeps resetting the timeout.

This setting should be a throttle rather than a debounce, imo.

**Comment by eXponenta** [comment]
But there not case when animated resize is required, if so, then this architecture apriory wrong.

**Comment by bigtimebuddy** [comment]
I think you're right @themoonrat. If I do this as only a throttle like I had before, users can still implement their own debounce with the `resize` event. That's probably the most minimal implementation.

File: packages/app/src/ResizePlugin.ts [hunk 2]
```diff
@@ -9,9 +9,13 @@ import { IApplicationOptions } from './Application';
  */
 export class ResizePlugin
```
Comment (cursedcoder) [inline packages/app/src/ResizePlugin.ts]: This plugin is hard to understand with all its static/this context mixed.

**Comment by cursedcoder** [comment]
Not sure if it's the right thing for the plugin to be singleton-ish.
Was there a reason?

File: packages/app/src/ResizePlugin.ts (hunk 2 above)
Comment (bigtimebuddy) [reply packages/app/src/ResizePlugin.ts]: yeah these are static methods where the context is the Application instance. Loader plugins are similar.

**Comment by bigtimebuddy** [comment]
It’s not really a singleton. No one really calls this class directly. It comes preinStalled, it’s mostly a code organization. Could’ve been done a bunch of ways, but this is how it evolved

**Pull request merged**